lib/cocoapods/user_interface.rb

```diff
@@ -331,13 +337,18 @@ def puts(message = '')
       #        The message to print.
       #
       def print(message)
-        STDOUT.print(message) unless config.silent?
+        return if config.silent?
+        begin
+          (io || STDOUT).print(message)
+        rescue Errno::EPIPE
+          exit 0
+        end
       end
 
       # gets input from $stdin
       #
       def gets
-        $stdin.gets
+        (io || $stdin).gets
```

Comment:
hmm, since most pipes used here won't be bidirectional, let's keep this as `stdin` for now?